### PR TITLE
Stratified splitter

### PR DESCRIPTION
**Pull request by marenpielka**
Implemented stratified splitter, using training_test_split from sklearn.
The data set will be split any number of times, according to the split_config (only "train" needs to be there, everything else is optional), and stratified with respect to the label distribution.

### CODE REVIEW

File: src/data_stack/dataset/splitter.py [hunk 1]
```diff
@@ -84,6 +89,42 @@ def get_indices(self, dataset_iterator: DatasetIteratorIF) -> List[List[int]]:
         return splits_indices
 
 
+class StratifiedSplitterImpl(SplitterIF):
+
+    def __init__(self, ratios: List[float]):
```
Comment (le1nux) [inline src/data_stack/dataset/splitter.py:94]: What about adding a `shuffle_seed: int = None` parameter to the constructor, to shuffle the split indices?
Otherwise, we always take the same samples in the same order.   

The `train_test_split` function can take a shuffle and a random state parameter:  https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html

**Review by le1nux — APPROVED** [review]
Hey @marenpielka 
nice work! :) We can merge this. What do you think about the shuffling remark?

File: src/data_stack/dataset/splitter.py (hunk 1 above)
Comment (marenpielka) [reply src/data_stack/dataset/splitter.py:94]: Sure, we can pass a seed parameter. train_test_split does the shuffling anyways if stratify=True, but seeding is not such a bad idea I guess :D
